tests/test_campaign.py

```diff
@@ -29,3 +35,32 @@ def test_get_surrogate(campaign, n_iterations, batch_size):
 
     model = campaign.get_surrogate()
     assert model is not None, "Something went wrong during surrogate model extraction."
+
+
+@pytest.mark.parametrize(
+    ("anti", "expected"),
```

Comment:
should this also run over the `exclude` argument?

Reply:
Have added it and also extended the test to actually verify if the metadata is updated correctly (without it, the effect of the `exclude` argument would not be visible in the test). Resolve if happy

Reply:
gucci